### PR TITLE
SONARJAVA-1839 Take into account referenced method arguments

### PR DESCRIPTION
**Pull request by m-g-sonar**

### CODE REVIEW

File: java-frontend/src/main/java/org/sonar/java/resolve/Resolve.java [hunk 1]
```diff
@@ -575,30 +578,35 @@ private boolean isArgumentsAcceptable(List<JavaType> argTypes, List<JavaType> fo
...
+      JavaType formal = formals.get(i);
+      if (!isAcceptableType(env, arg, formal, autoboxing)) {
         return false;
       }
     }
     return true;
   }
 
-  private boolean isAcceptableType(JavaType arg, JavaType formal, boolean autoboxing) {
+  private boolean isAcceptableType(Env env, JavaType arg, JavaType formal, boolean autoboxing) {
```
Comment (SonarTech) [inline java-frontend/src/main/java/org/sonar/java/resolve/Resolve.java]: MAJOR Reduce the number of returns of this method 6, down to the maximum allowed 5. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS1142)

**Comment by SonarTech** [comment]
SonarQube analysis reported 1 issue
* MAJOR 1 major

Watch the comments in this conversation to review them.

File: java-frontend/src/main/java/org/sonar/java/resolve/TypeAndReferenceSolver.java [hunk 2]
```diff
@@ -556,15 +556,19 @@ private void refineLambdaType(LambdaExpressionTreeImpl lambdaExpressionTree, Jav
...
       // found a lambda return type different from the one infered : update infered type
-      TypeSubstitution typeSubstitution = ((ParametrizedTypeJavaType) expressionType).typeSubstitution;
-      typeSubstitution.substitutionEntries().stream()
+      if (expressionType.isTagged(JavaType.PARAMETERIZED)) {
+        TypeSubstitution typeSubstitution = ((ParametrizedTypeJavaType) expressionType).typeSubstitution;
+        typeSubstitution.substitutionEntries().stream()
         .filter(e -> e.getValue() == capturedReturnType)
         .findFirst()
         .ifPresent(e -> {
           TypeSubstitution refinedSubstitution = new TypeSubstitution(typeSubstitution).add(e.getKey(), refinedReturnType);
```
Comment (SonarTech) [inline java-frontend/src/main/java/org/sonar/java/resolve/TypeAndReferenceSolver.java]: MAJOR Make this line start at column 13. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AIndentationCheck)

**Comment by SonarTech** [comment]
SonarQube analysis reported 2 issues
* MAJOR 2 major

Watch the comments in this conversation to review them.

File: java-frontend/src/main/java/org/sonar/java/resolve/TypeAndReferenceSolver.java (hunk 2 above)
Comment (m-g-sonar) [inline java-frontend/src/main/java/org/sonar/java/resolve/TypeAndReferenceSolver.java]: FP

**Pull request merged**